### PR TITLE
Shoot capacity & death system

### PR DESCRIPTION
**Pull request by raphaelMrci**
## Description
Add players death, shoot capacity & enemy death

## Lien tache Jira
[lien du/des ticket Jira]
https://leomaman.atlassian.net/browse/TYPE-57?atlOrigin=eyJpIjoiMzMzZWJkNGUyNjFhNGQ1ZjkyZTY1MzZiZDk1N2E5ZmEiLCJwIjoiaiJ9

## Changement
- player shoot
- player death
- enemy spawn from server
- enemy killable
- player bullets

### Summary by CodeRabbit

- New Feature: Added automated checks for code documentation and formatting in the GitHub Actions workflow.
- New Feature: Expanded the game's Entity Component System (ECS) with new components and entities, enhancing gameplay possibilities.
- New Feature: Improved network packet management, including the addition of new packet types and better error handling.
- New Feature: Introduced a new game system for handling game logic and events.
- Refactor: Renamed and reorganized several files and classes for better code clarity and organization.
- Bug Fix: Enhanced collision detection logic for better game performance.
- Style: Updated variable naming conventions for consistency across the codebase.
- Chore: Updated the project's CMakeLists.txt file to include new and renamed files.

### CODE REVIEW

**Review by christophechr — APPROVED** [review]

File: client/ecs/components/ControlComponent.cpp [hunk 1]
```diff
@@ -29,7 +29,6 @@ void ECS::ControlComponent::execute(std::vector<Network::Packet> &packetsQueue,
     if (sf::Keyboard::isKeyPressed(sf::Keyboard::Down)) {
         packet = Network::PacketManager::createPacket(Network::PacketType::MOVE_DOWN);
         packetsQueue.push_back(*packet);
-
     }
     if (sf::Keyboard::isKeyPressed(sf::Keyboard::Left)) {
         packet = Network::PacketManager::createPacket(Network::PacketType::MOVE_LEFT);
```
[the remark this answers is not in the document]
Comment (raphaelMrci) [reply client/ecs/components/ControlComponent.cpp:34]: Already done

File: client/ecs/Core.cpp
Comment (STCB) [inline client/ecs/Core.cpp:1]: J'approuve le double include

File: client/ecs/components/ClickComponent.cpp
Comment (STCB) [inline client/ecs/components/ClickComponent.cpp:1]: `if (sf::Mouse::isButtonPressed(sf::Mouse::Left)) {
        if (_rect.contains(mousePos)) {
            if (_callback)
                _callback(packetsQueue, entity);`

Pas top comme notation, plutot mettre

`if (sf::Mouse::isButtonPressed(sf::Mouse::Left)) {
        if (_rect.contains(mousePos) && _callback) {
            _callback(packetsQueue, entity);`

voire

``if (sf::Mouse::isButtonPressed(sf::Mouse::Left) && _rect.contains(mousePos) && _callback) {
      _callback(packetsQueue, entity);``

File: client/ecs/components/SpriteComponent.cpp [hunk 2]
```diff
@@ -31,7 +31,31 @@ std::shared_ptr<ECS::IComponent> ECS::SpriteComponent::clone() const
...
+        } else {
+            _rect.top += _rect.height;
+            _iterations.y++;
+        }
+    } else {
+        _rect.left += _rect.width;
+        _iterations.x++;
+    }
+    _sprite.setTextureRect(_rect);
+}
```
[the remark this answers is not in the document]
Comment (STCB) [reply client/ecs/components/SpriteComponent.cpp:61]: pas con le bougre

File: client/ecs/models/EnemyEntity.cpp [hunk 3]
```diff
@@ -11,27 +11,50 @@
...
+    EnemyEntity::EnemyEntity(int id) : Entity(id)
+    {
+        addComponent(std::make_shared<ECS::PositionComponent>(0, 0));
+        addComponent(std::make_shared<ECS::ScaleComponent>(0.03f, 0.03f));
+        // addComponent(std::make_shared<ECS::RotationComponent>(270.0f));
+
+        sf::Texture enemyTexture;
+        if (!enemyTexture.loadFromFile(ConvertPath::convertPath("assets/Ship5.png"))) {
+            std::cout << "Error loading enemy texture" << std::endl;
+            return;
```
[the remark this answers is not in the document]
Comment (STCB) [reply client/ecs/models/EnemyEntity.cpp:27]: @raphaelMrci std::cerr

File: server/ecs/components/HitboxComponent.hpp [hunk 4]
```diff
@@ -12,6 +12,7 @@
 #include "Entity.hpp"
 #include "Packet.hpp"
 
+
```
Comment (STCB) [inline server/ecs/components/HitboxComponent.hpp:15]: ?

**Review by STCB — COMMENTED** [review]
Quelques petites modifs qui prennent 2 minutes

File: ecs/system/GameSystem.cpp [hunk 5]
```diff
@@ -18,7 +18,18 @@
...
+    for (auto entity : actualScene->entitiesList) {
+        if (entity == nullptr || !entity->isEnabled)
+            continue;
+        std::vector<std::shared_ptr<ECS::AGameComponent>> gameComponents = entity->getComponents<ECS::AGameComponent>();
+
+        for (auto component : gameComponents) {
+            if (component != nullptr)
+                component->update(packetQueue, *entity, deltaTime);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (raphaelMrci) [reply ecs/system/GameSystem.cpp:34]: How could I fix those performance issues ?

File: server/ecs/components/HitboxComponent.hpp (hunk 4 above)
Comment (raphaelMrci) [reply server/ecs/components/HitboxComponent.hpp:15]: don't know why lol

File: install.sh [hunk 6]
```diff
@@ -32,8 +32,8 @@ install_yum() {
 }
 
 install_pacman() {
-    sudo pacman -S g++
-    sudo pacman -S mesa lib32-mesa wine-staging winetricks gdiplus freetype2 openal flac libvorbis libvorbisfile libvorbisenc libogg wine-mono wine-gecko libudev
+    sudo pacman -S g++ --noconfirm
```
Comment (mangasteak) [inline install.sh:35]: a quoi ça sert ?

File: server/ecs/CollisionSystem.cpp [hunk 7]
```diff
@@ -54,13 +54,18 @@ namespace ECS {
...
                 int _y2 = pos2[1] + hitbox2[1].second;
-/*
-                std::cout << "Trying to collide " << entity->getId() << " with " << entity2->getId() << std::endl;
-                std::cout << (_x1 > x1 && _x1 < x2 && _y1 > y1 && _y1 < y2) << std::endl;
-                std::cout << (_x2 > x1 && _x2 < x2 && _y1 > y1 && _y1 < y2) << std::endl;
-                std::cout << (_x1 > x1 && _x1 < x2 && _y2 > y1 && _y2 < y2) << std::endl;
-                std::cout << (_x2 > x1 && _x2 < x2 && _y2 > y1 && _y2 < y2) << std::endl;
-                std::cout << "---------------------" << std::endl;
+
+                if (entity->getId() == 4 && entity2->getId() == 5) {
```
Comment (mangasteak) [inline server/ecs/CollisionSystem.cpp:58]: pourquoi print les coordonées ? pour debug ? pourquoi push des print de debug si c est le cas ?

**Review by mangasteak — APPROVED** [review]
lit mes commentaires avant de merge stp

File: install.sh (hunk 6 above)
Comment (raphaelMrci) [reply install.sh:35]: a eviter de devoir appuyer sur yes quand l'installer se lance, genre equivalente -y dans d'autres pckmngr

**Pull request merged**